NAudio/Midi/MidiEvent.cs

```diff
@@ -192,6 +192,13 @@ public MidiEvent(long absoluteTime, int channel, MidiCommandCode commandCode)
         }
 
         /// <summary>
+        /// Creates a deep clone of this MIDI event.
+        /// </summary>
+        public virtual MidiEvent Clone() => (MidiEvent)MemberwiseClone();

```

Comment:
I'd prefer to make this abstract so that the onus is always on the subclass to figure out if it's doing the right thing to clone. That would require making MidiEvent abstract which would require either introducing an UnknownMidiEvent or making sure all midi events are accounted for or throwing an exception on unknown event types.